prow/scripts/cluster-integration/kyma-cli-alpha-uninstall.sh

```diff
@@ -0,0 +1,184 @@
+    GARDENER_KYMA_PROW_PROVIDER_SECRET_NAME
+)
+
+utils::check_required_vars "${requiredVars[@]}"
+
+# nice cleanup on exit, be it succesful or on fail
+trap gardener::cleanup EXIT INT
+
+RANDOM_NAME_SUFFIX=$(LC_ALL=C tr -dc 'a-z0-9' < /dev/urandom | head -c4)
+readonly COMMON_NAME_PREFIX="grdnr"
+COMMON_NAME=$(echo "${COMMON_NAME_PREFIX}${RANDOM_NAME_SUFFIX}" | tr "[:upper:]" "[:lower:]")
+
+### Cluster name must be less than 10 characters!
+export CLUSTER_NAME="${COMMON_NAME}"
+
+# Local variables
+
+#Used to detect errors for logging purposes
+ERROR_LOGGING_GUARD="true"
+
+log::info "Building Kyma CLI"
+cd "${KYMA_PROJECT_DIR}/cli"
+make build-linux
+mv "${KYMA_PROJECT_DIR}/cli/bin/kyma-linux" "${KYMA_PROJECT_DIR}/cli/bin/kyma"
+export PATH="${KYMA_PROJECT_DIR}/cli/bin:${PATH}"
+
+log::info "Provision cluster: \"${CLUSTER_NAME}\""
+
+gardener::set_machine_type
+
+gardener::provision_cluster
+
+log::info "Installing Kyma"
+
+# Parallel-install library installs cluster-essentials, istio, and xip-patch before kyma installation. That's why they should not exist on the InstallationCR.
+# Once we figure out a way to fix this, this custom CR can be deleted from this script.
+cat << EOF > "/tmp/kyma-parallel-install-installationCR.yaml"
+apiVersion: "installer.kyma-project.io/v1alpha1"
+kind: Installation
+metadata:
+  name: kyma-installation
+  namespace: default
+spec:
+  components:
+    - name: "testing"
+      namespace: "kyma-system"
+    - name: "knative-eventing"
+      namespace: "knative-eventing"
+    - name: "dex"
+      namespace: "kyma-system"
+    - name: "ory"
+      namespace: "kyma-system"
+    - name: "api-gateway"
+      namespace: "kyma-system"
+    - name: "rafter"
+      namespace: "kyma-system"
+    - name: "service-catalog"
+      namespace: "kyma-system"
+    - name: "service-catalog-addons"
+      namespace: "kyma-system"
+    - name: "helm-broker"
+      namespace: "kyma-system"
+    - name: "nats-streaming"
+      namespace: "natss"
+    - name: "core"
+      namespace: "kyma-system"
+    - name: "cluster-users"
+      namespace: "kyma-system"
+    - name: "logging"
+      namespace: "kyma-system"
+    - name: "permission-controller"
+      namespace: "kyma-system"
+    - name: "apiserver-proxy"
+      namespace: "kyma-system"
+    - name: "iam-kubeconfig-service"
+      namespace: "kyma-system"
+    - name: "serverless"
+      namespace: "kyma-system"
+    - name: "knative-provisioner-natss"
+      namespace: "knative-eventing"
+    - name: "event-sources"
+      namespace: "kyma-system"
+    - name: "application-connector"
+      namespace: "kyma-integration"
+    - name: "tracing"
+      namespace: "kyma-system"
+    - name: "monitoring"
+      namespace: "kyma-system"
+    - name: "kiali"
+      namespace: "kyma-system"
+    - name: "console"
+      namespace: "kyma-system"
+EOF
+
+(
+cd "${KYMA_PROJECT_DIR}/kyma"
+cli-alpha:deploy "${KYMA_PROJECT_DIR}/kyma/resources" "/tmp/kyma-parallel-install-installationCR.yaml"
```

Comment:
Instead of creating it that way I was rather suggesting creating a function called `deploy_kyma` inside this script which would do the following:
```sh
function deploy_kyma() {
  kyma alpha deploy \
  --ci \
  --resources "${KYMA_PROJECT_DIR}/kyma/resources" \
  --components "/tmp/kyma-parallel-install-installationCR.yaml"
}
```
If this is something in the early development stage and is not used by multiple cloud providers I would go against creating a new library. Just a simple function would be enough just not to copy the code that can be centralised.